pkg/core/app/app.go

```diff
@@ -69,12 +69,15 @@ func (a *App) Setup(_ context.Context) error {
 	}
 	a.docker = d
 
-	if (a.kind == utils.Docker || a.kind == utils.DockerCompose) && IsDetachMode(a.cmd) {
-		return fmt.Errorf("detach mode is not allowed in Keploy command")
+	if a.kind == utils.DockerStart || a.kind == utils.DockerRun || a.kind == utils.DockerCompose {
+		err = IsDetachMode(a.cmd, a.kind)
```

Comment:
`IsDetachMode` should return bool value only and log the message here only.

Reply:
Refactored